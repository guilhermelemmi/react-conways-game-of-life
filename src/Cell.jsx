import React from 'react';

const Cell = ({ alive }) => (
  <li className={`cell ${alive ? 'cell-alive' : ''}`} />
);

export default Cell;
