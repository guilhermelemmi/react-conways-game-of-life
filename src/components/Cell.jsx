import React from 'react';

const Cell = ({ alive }) => (
  <div className={alive ? 'cell cell-alive' : 'cell'} />
);

export default Cell;
