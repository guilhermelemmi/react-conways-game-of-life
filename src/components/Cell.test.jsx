import React from 'react';
import renderer from 'react-test-renderer';

import Cell from './Cell';

describe('<Cell />', () => {
  it('should match snapshot', () => {
    const cell = renderer.create(
      <Cell />
    ).toJSON();
    expect(cell).toMatchSnapshot();
  });

  it('should match alive snapshot', () => {
    const cell = renderer.create(
      <Cell alive/>
    ).toJSON();
    expect(cell).toMatchSnapshot();
  });
});
