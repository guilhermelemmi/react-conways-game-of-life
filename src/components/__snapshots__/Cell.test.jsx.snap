// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cell /> should match alive snapshot 1`] = `
<li
  className="cell cell-alive"
/>
`;

exports[`<Cell /> should match snapshot 1`] = `
<li
  className="cell "
/>
`;
